### PR TITLE
Make the CDI Config more resilient to runtime errors

### LINKED ISSUE #674: Installing, removing, and reinstalling causes the controller to crash
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
kind bug
> /kind enhancement

**What happened**:

In controller log I see:

`
E0226 01:26:09.316306       1 config-controller.go:319] Error updating CDI Config: resource name may not be empty
F0226 01:26:09.316993       1 controller.go:220] Error running config controller: %+v Error creating CDI config: Error updating CDI Config: resource name may not be empty
goroutine 77 [running]:
kubevirt.io/containerized-data-importer/vendor/github.com/golang/glog.stacks(0xc42047c800, 0xc42034e000, 0xab, 0xe4)
	/go/src/kubevirt.io/containerized-data-importer/vendor/github.com/golang/glog/glog.go:769 +0xcf
kubevirt.io/containerized-data-importer/vendor/github.com/golang/glog.(*loggingT).output(0x1b6f7e0, 0xc400000003, 0xc4200f80b0, 0x1ad590d, 0xd, 0xdc, 0x0)
	/go/src/kubevirt.io/containerized-data-importer/vendor/github.com/golang/glog/glog.go:720 +0x32d
kubevirt.io/containerized-data-importer/vendor/github.com/golang/glog.(*loggingT).println(0x1b6f7e0, 0x3, 0xc420723f98, 0x2, 0x2)
	/go/src/kubevirt.io/containerized-data-importer/vendor/github.com/golang/glog/glog.go:633 +0xe7
kubevirt.io/containerized-data-importer/vendor/github.com/golang/glog.Fatalln(0xc420723f98, 0x2, 0x2)
	/go/src/kubevirt.io/containerized-data-importer/vendor/github.com/golang/glog/glog.go:1141 +0x53
main.start.func8(0xc4204020f0, 0xc420098780, 0xc4203a2820)
	/go/src/kubevirt.io/containerized-data-importer/cmd/cdi-controller/controller.go:220 +0xc3
created by main.start
	/go/src/kubevirt.io/containerized-data-importer/cmd/cdi-controller/controller.go:217 +0xd70
`

**What you expected to happen**:  no errors

**How to reproduce it (as minimally and precisely as possible)**:

```bash
make cluster-up cluster-sync
./cluster/kubectl.sh delete cdi cdi
./cluster/kubectl.sh apply -f manifests/generated/cdi-operator-cr.yaml
```

**Anything else we need to know?**:

Couple things to note:  The CDIConfig object is not removed when the cdi object is deleted which is weird.  I think that's actually a Kubernetes bug that is fixed in 1.12 or 1.13.

The CDIConfig object should probably be owned by the CDI obect.  See `SetOwner` in pkg/operator.  

**Environment**:
- CDI version (use `kubectl get deployments cdi-deployment -o yaml`): f025943d706dfeed9620c5a2c771598e93320520
- Kubernetes version (use `kubectl version`): kubernetes 1.11.0
- Cloud provider or hardware configuration:
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by igoihman**
Signed-off-by: Irit goihman <igoihman@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #674

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by igoihman** [comment]
This is strange, how did you see it doesn't have CDI label?  I'm creating it with this label: `app: containerized-data-importer`
The issue was caused because there was an attempt to create CDI Config with an empty name.
I'm trying to find the cause for that.

**Comment by igoihman** [comment]
The root cause of the error consists of two parts:
1. CDI Config was missing the owner (CDI), so when CDI was deleted, CDI Config object remained there.
2. k8s 1.11.0 bug: the config was there and when I ran `./cluster/kubectl.sh describe cdiconfig config -n cdi` I could see it and its fields, so when we attempted to re-create it we get `IsAlreadyExists` error and an empty CDI Config object from k8s. When we attempted to update an object without a name we hit this error.
Now when we remove CDI the object is also removed.

File: pkg/controller/util.go [hunk 1]
```diff
@@ -937,6 +944,7 @@ func MakeCDIConfigSpec(name string) *cdiv1.CDIConfig {
 			Name: name,
 			Labels: map[string]string{
 				common.CDILabelKey: common.CDILabelValue,
+				"cdi.kubevirt.io":  "",
```
Comment (mhenriks) [inline pkg/controller/util.go]: should probably use `common.CDIComponentLabel` here

**Comment by mhenriks** [comment]
Ran into #674 yesterday during functional test run for #676.  So it would seem to be possible to get in into that state without remove/reinstall.  So I think point 2 you mention above should be addressed in this PR.  Let's not just rely on the CDIConfig to not be there.

**Comment by awels** [comment]
@mhenriks This good to go now?

**Comment by mhenriks** [comment]
@awels unless I'm missing something I don't see point 2 above addressed.

**Comment by igoihman** [comment]
@mhenriks second commit addresses it. I'm calling update with the config spec I created and not with the object that was returned from `create`.

**Pull request merged**